Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -4369,6 +4402,214 @@ HTTPServer server({
+    RetainPtr manipulationConfiguration = adoptNS([[_WKTextManipulationConfiguration alloc] init]);
+    manipulationConfiguration.get().includeSubframes = YES;
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://example.com/example"]]];
+    Util::run(&receivedMessage1);
+
+    [webView _startTextManipulationsWithConfiguration:manipulationConfiguration.get() completion:^{ }];
+    while ([textManipulationDelegate items].count < 2)
+        Util::spinRunLoop();
+
+    RetainPtr items = [textManipulationDelegate items];
+    auto sortFunction = ^(_WKTextManipulationItem *item1, _WKTextManipulationItem *item2) {
+        auto value1 = [NSNumber numberWithBool:item1.isSubframe];
+        auto value2 = [NSNumber numberWithBool:item2.isSubframe];
+        return [value1 compare:value2];
+    };
+    RetainPtr sortedItems = [items.get() sortedArrayUsingComparator:sortFunction];
+    EXPECT_EQ(items.get().count, 2UL);
+    auto firstItem = [sortedItems objectAtIndex:0];
+    auto secondItem = [sortedItems objectAtIndex:1];
+    EXPECT_EQ(firstItem.isSubframe, NO);
+    EXPECT_EQ(firstItem.isCrossSiteSubframe, NO);
+    EXPECT_EQ(firstItem.tokens.count, 1UL);
+    EXPECT_STREQ("mainframe content", firstItem.tokens[0].content.UTF8String);
+    EXPECT_EQ(secondItem.isSubframe, YES);
+    EXPECT_EQ(secondItem.isCrossSiteSubframe, YES);
+    EXPECT_EQ(secondItem.tokens.count, 1UL);
+    EXPECT_STREQ("iframe content", secondItem.tokens[0].content.UTF8String);
+
+    __block bool done = false;
+    [webView _completeTextManipulationForItems:@[
+        (_WKTextManipulationItem *)createItem(firstItem.identifier, { createToken(firstItem.tokens[0].identifier, @"MAINFRAME CONTENT") }),
+        (_WKTextManipulationItem *)createItem(secondItem.identifier, { createToken(secondItem.tokens[0].identifier, @"IFRAME CONTENT") })
+    ] completion:^(NSArray<NSError *> *errors) {
+        EXPECT_EQ(errors, nil);
+        done = true;
+    }];
+    Util::run(&done);
+
+    [webView evaluateJavaScript:@"getTextContent()" completionHandler:nil];
+    Util::run(&receivedMessage2);
+
+    [webView evaluateJavaScript:@"getIframeTextContent()" completionHandler:nil];
+    Util::run(&receivedMessage3);
+}
+
+TEST(SiteIsolation, CompleteTextManipulationFailsInSomeFrame)
+{
+    static constexpr auto mainFrameBytes = R"TESTRESOURCE(
+    <div>mainframe content</div>
+    <script>
+        function removeIframe() {
+            let element = document.getElementById('iframe');
+            element.parentNode.removeChild(element);
+        }
+        var messageCount = 0;
+        addEventListener('message', () => {
+            if (++messageCount == 2)
+                window.webkit.messageHandlers.testHandler.postMessage('loaded');
+        }, false);
+    </script>
+    <iframe id='iframe' src='https://apple.com/apple'></iframe>
+    <iframe src='https://apple.com/apple'></iframe>
```

Comment:
Doesn't this make two iframes share the process since they're both accessing https://apple.com?
We ideally have another test case or modify this one to have two distinct cross-site addresses.